pkg/sysinfo/sysinfo_darwin.go

```diff
@@ -19,6 +23,11 @@ var (
 
 // OSVersion returns the system's OS version.
 func OSVersion() (*OSVersionInfo, error) {
+	cached, found := sysinfoCache.Get(osVersionInfoCacheKey)
+	if found {
```

Comment:
```suggestion
	if cached, found := sysinfoCache.Get(osVersionInfoCacheKey); found {
```

Fwiw; this is the shorthand version. Not asking for change so much as I'm sharing for future reference, should you want to use it.

Reply:
As I was writing, I was wondering if Go had this feature. I'm glad it does :) I'll make the change because it'll get me in the mindset of doing this.